resolve/index.go

```diff
@@ -109,9 +112,16 @@ func NewRuleIndex(kindToResolver map[string]Resolver) *RuleIndex {
 //
 // AddRule may only be called before Finish.
 func (ix *RuleIndex) AddRule(c *config.Config, r *rule.Rule, f *rule.File) {
-	var imps []ImportSpec
-	if rslv, ok := ix.kindToResolver[r.Kind()]; ok {
+	var (
+		imps []ImportSpec
+		rslv Resolver
+		ok   bool
+	)
+	if rslv, ok = ix.kindToResolver[r.Kind()]; ok {
```

Comment:
I think we should replace `RuleIndex.kindToResolver` with a callback of type `func(r *rule.Rule, f *rule.File) Resolver` implemented in fix-update.go. It will use `r.Kind()` and `f.Pkg`.

This should make the `rule.resolver` field and `RuleIndex.Resolver` method unnecessary.

Reply:
OK - Note that FindRulesByImport does not have a relevant *rule.File to pass to kindToResolver, but that can work out if we specify that a nil *rule.File is permitted but ignores any replacements.

Reply:
I implemented a version that used a regular func, but in the course of cleaning it up I thought it was better to have a type for it instead. Otherwise it was passing in a method of type, which is a little unusual, and I didn't perceive a big advantage of having RuleIndex avoid the explicit dependency on that type.